alarm_trap_sender.cpp

```diff
@@ -138,8 +142,40 @@ unsigned long AlarmFilter::current_time_ms()
   return ts.tv_sec * 1000 + (ts.tv_nsec / 1000000);
 }
 
+alarmActiveTable_SNMPDateTime* AlarmTrapSender::alarm_time_issued(void)
+{
+  alarmActiveTable_SNMPDateTime* d_time = new alarmActiveTable_SNMPDateTime();
+  struct timespec ts;
+  struct tm *timing;
+  time_t rawtime;
+  char plus_or_minus;
+  int daylight_savings;
+
+  clock_gettime(CLOCK_REALTIME, &ts);
+  rawtime = ts.tv_sec;
+  timing = gmtime(&rawtime);
+  // This updates the variable timezone declared within time.h which
+  // tells us how offset (in seconds) from UTC we are.
+  localtime(&rawtime);
```

Comment:
I think you should be setting `timing` from `localtime`, not from `gmtime` - otherwise you're using the UTC hours and seconds, even when you're specifying an offset from UTC.